### ISSUE TITLE
BUG - 3 - All -Benefit summary and service verification letter stuck showing loading component when beneficiary data request fails

### ISSUE THREAD

**Issue by theodur** [department-of-veterans-affairs/va-mobile-app#6951]
## What happened?

When the request for fetching beneficiary data fails, the `Benefit summary and service verification letter` screen is stuck showing the loading component. This is caused by the [logic](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/123a0b450054423581e953e65874a962747f2958/VAMobile/src/screens/BenefitsScreen/Letters/BenefitSummaryServiceVerification/BenefitSummaryServiceVerification.tsx#L217-L223) in the `onViewLetter` function that gets called when a user tries to view the letter.

It's not clear whether we should show an error if the beneficiary data request fails, as the logic for generating the letter will still generate the letter if beneficiary data isn't available. It needs to be determined whether we should show an error when the request fails, or if we should still allow the user to view the letter without the beneficiary data included.

## Specs:

- Device:
- OS:
- User Account:
- Accessibility State:

## Steps to Reproduce

- Throw an error in the `getLetterBeneficiaryData` thunk in the `lettersSlice` or mock an error from the `/v0/letters/beneficiary` endpoint with Charles Proxy.

- Navigate to the Benefits -> VA letters and documents -> Review letters -> Benefit summary and service verification letter

- Notice the loading component is stuck on the screen

## Desired behavior

The Benefit summary and service verification letter should not be stuck on the loading component if the beneficiary data request fails.

## Acceptance Criteria

- [ ] Determine whether an error should be displayed or the user should still be allowed to see the letter without the beneficiary data present
- [ ] Make necessary updates to handle failed beneficiary data in the correct way

## Bug Severity - BE SURE TO ADD THE SEVERITY LABEL
 
See [Bug Tracking](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/QA#issue-severity) for details on severity levels

- Impact: High Low 
- Frequency: High Low 

- 3 - Low
- 2 - High
- 1 - Critical

## Screen shot(s) and additional information

<details>
  <summary>Full JSON response for services related to issue (expand/collapse)</summary>
  

</details>

## Ticket Checklist
- [ ] Steps to reproduce are defined
- [ ] Desired behavior is added
- [ ] Labels added (front-end, back-end, relevant feature, accessibility, etc)
- [ ] Estimate of 1 added (for front-end tickets)
- [ ] Added either to the relevant feature epic (if found during new feature implementation), or the [bug epic](https://app.zenhub.com/workspaces/va-mobile-product-view-610035bc5395bb000e62e529/issues/gh/department-of-veterans-affairs/va-mobile-app/5367)

**Comment by TKDickson** [comment]
Hey @bischoffa - we're going to need UX decision on this part (below). I've seen a few bugs where there's been a splitting off UX ticket for that decision-making just recently, is there a formal process for that (who should spin up the UX ticket, where the bug ticket goes in teh meantime, ticket templates) that you know of? Should I ask a UX person instead?

**Comment by TKDickson** [comment]
@bischoffa sounds good! Will add the blocked-UX label and move to the blocked column (after the other scrub stuff is done) when we see them. That separate UX decision ticket is a good idea.

**Comment by timwright12** [comment]
@bischoffa @TKDickson any news on this one? Been blocked by UX for a while

**Comment by Sparowhawk** [comment]
@wavelaurenrussell if you could respond in this ticket instead of the other one that would be great lol

**Comment by timwright12** [comment]
@wavelaurenrussell @ala-yna ping for response on this ticket, seems blocked by UX

**Comment by wavelaurenrussell** [comment]
Can someone explain why this has a `Blocked by UX` label? What is the block?

**Comment by timwright12** [comment]
@wavelaurenrussell  I think it's the second paragraph in the description @TKDickson called out:

> It's not clear whether we should show an error if the beneficiary data request fails, as the logic for generating the letter will still generate the letter if beneficiary data isn't available. It needs to be determined whether we should show an error when the request fails, or if we should still allow the user to view the letter without the beneficiary data included.

**Comment by TKDickson** [comment]
The screen you're stuck on forever, if the beneficiary call errors out (never resolves into an error state): ![Image](https://github.com/user-attachments/assets/ce8bd907-7b3b-44f3-a1bb-e667864f06ff)

**Comment by wavelaurenrussell** [comment]
Perfect thanks! I think the solution should still allow the beneficiary travel screen to load so that users can get their letter. And then we can present an Alert at the top that describes that the failure happened; _you can still get your letter but things may be inaccurate_.

Would this cover it? If so, I'll loop in Misty next.

**Comment by TKDickson** [comment]
@wavelaurenrussell yes it would. I *think* what would happen is you'd be able to generate a letter (with the right information), but not be able to pick what shows on it (aka it would default to showing all information) 

That's a ~80% guess, but after this UI/front-end thing is resolved I could confirm it.

**Comment by timwright12** [comment]
@JonathanPostMO can you read through the previous team's UX recommendation for this issue and either approve or change it how you would prefer this to be?

@BeccaTupaj thoughts on this one for @AdryienH?

**Comment by JonathanPostMO** [comment]
@timwright12 I think we can handle this using a mix of inline alerts (i.e. the new component we're working on for "empty" states) and/or toasts (snackbar)—similar to what the previous UX team was on to, but more contextual and with a "try again" option in the toast that allows the user to try and recover from the error by pinging the API to try loading again. Is this last bit technical feasible today though, could need extra BE work?

**Comment by timwright12** [comment]
@dumathane Do we have a component that does what @JonathanPostMO mentioned?

**Comment by JonathanPostMO** [comment]
@timwright12 We'd need to update this component: https://department-of-veterans-affairs.github.io/va-mobile-app/design/Components/Alerts%20and%20progress/Snackbar/ @dumathane Does this documentation match what's in prod?

**Comment by timwright12** [comment]
If we have something already built, we can certainly implement, but if a component needs alteration we should move that scope to another ticket.

**Comment by JonathanPostMO** [comment]
@BeccaTupaj Could we open a design exploration ticket for toast updates? I had already discussed this with Ryan (updating the design to be more semantic, using colors & icons to create more states). Could we also figure out a way to maybe roll out an MVP to bandage this bug, and follow up with a "_More_ ideal UX" implementation once updated toasts are ready for prod?

**Comment by BeccaTupaj** [comment]
@JonathanPostMO For resolving this bug ticket we can use the alert component as it exists now. I can spin up a separate exploration for toast/snackbar updates and agree this could be a good place for implementing, but for this bug (and bugs in general) let's respond with the most direct and ready to use solution. 

It sounds like the solution needed for this bug is as follows: 
1) An alert at the top of the page using the current snackbar component.
2) Letter loads without beneficiary data included. 

@timwright12 I dont think this could go directly to @AdryienH. It will need a mockup and the content of the alert will need to be confirmed along with the decision to load the letter without beneficiary data. (cc: @rtwell Do you know who would be the right PO to confirm content of alert and decision to load the letter?)

**Comment by JonathanPostMO** [comment]
@BeccaTupaj Minor update to the above: An alert at the top of the page using the current _error alert_ [component](https://department-of-veterans-affairs.github.io/va-mobile-app/design/Components/Alerts%20and%20progress/Alert#error). Sounds good too!

**Comment by timwright12** [comment]
Frontend estimate: 5pts

**Comment by BeccaTupaj** [comment]
@timwright12 This ticket seems more like a new error state than a bug. It seems like the app may be operating as expected per the code, and this error state was never built. Do you have more insight on that? (either way doesn't change the ticket content, just helps strategize the appropriate response from our team) .

**Comment by BeccaTupaj** [comment]
**Bug solution:** Generate an alert at the top of the screen when the beneficiary data does not load correctly. This way the user will not be stuck on a loading screen. They will still be able to see their letter, but will see an alert stating that there was an error when loading. 

**Content to use for alert:** 
Title: 
"Error when loading letter details"
Body:
"You can still get your letter but some beneficiary letter details may be inaccurate. Please try loading again to resolve error."

**Comment by JonathanPostMO** [comment]
@BeccaTupaj Could we rewrite the alert content so it's in [active voice](https://dev-design.va.gov/3904/content-style-guide/plain-language/use-active-voice)?
